pkg/agent/provisioning.go

```diff
@@ -1,24 +1,77 @@
 package agent
 
 import (
+	"fmt"
+
 	"github.com/Axway/agent-sdk/pkg/agent/handler"
 	v1 "github.com/Axway/agent-sdk/pkg/apic/apiserver/models/api/v1"
 	"github.com/Axway/agent-sdk/pkg/apic/apiserver/models/management/v1alpha1"
+	mv1a "github.com/Axway/agent-sdk/pkg/apic/apiserver/models/management/v1alpha1"
 	"github.com/Axway/agent-sdk/pkg/apic/provisioning"
+	"github.com/Axway/agent-sdk/pkg/migrate"
+	"github.com/Axway/agent-sdk/pkg/util/log"
 )
 
 // credential request definitions
 // createOrUpdateDefinition -
-func createOrUpdateDefinition(data v1.Interface) (*v1.ResourceInstance, error) {
+func createOrUpdateDefinition(data v1.Interface, marketplaceMigration migrate.Migrator) (*v1.ResourceInstance, error) {
 	if agent.agentFeaturesCfg == nil || !agent.agentFeaturesCfg.MarketplaceProvisioningEnabled() {
 		return nil, nil
 	}
-	return agent.apicClient.CreateOrUpdateResource(data)
+
+	var runMarketplaceMigration = false
+
+	// Check (only) credential request definition to see if it exists prior to CreateOrUpdateResource call
+	ri, err := data.AsInstance()
+	if mv1a.CredentialRequestDefinitionGVK().Kind == ri.Kind {
```

Comment:
if in fact we do only care about market place migration then this can be moved into the createOrUpdateCredentialRequestDefinition func.

Reply:
Good point.  Will do

Reply:
Moved to its own func...